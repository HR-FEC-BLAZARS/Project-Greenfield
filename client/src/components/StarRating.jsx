import React from 'react';

function getStars(value) {
  let starIcon = {
    wholeStar: (
      <span aria-hidden="true" className="full-star">
        ★
      </span>
    ),
    threeFourthStar: (
      <span aria-hidden="true" className="three-Fourth-star">
        ★
      </span>
    ),
    halfStar: (
      <span aria-hidden="true" className="half-star">
        ★
      </span>
    ),
    oneThirdStar: (
      <span aria-hidden="true" className="one-third-star">
        ★
      </span>
    ),
    emptyStar: (
      <span aria-hidden="true" className="empty-star">
        ★
      </span>
    ),
  };

  const stars = [];
  const [whole, part] = value ? value.toString().split('.') : [0, 0];

  for (let i = 0; i < whole; i++) {
    stars.push(starIcon.wholeStar);
  }

  if (part) {
    let num = part.toString();
    if (num >= 7.5) {
      stars.push(starIcon.threeFourthStar);
    } else if (num >= 5) {
      stars.push(starIcon.halfStar);
    } else if (num >= 2.5) {
      stars.push(starIcon.push(star.oneThirdStar));
    }
  }

  for (let j = whole; j < (part ? 4 : 5); j++) {
    stars.push(starIcon.emptyStar);
  }

  return stars;
}

const StarRating = ({ value }) => {
  return (
    <span className="stars">
      {getStars(value).map((star, index) => (
        <span key={index}>{star}</span>
      ))}
    </span>
  );
};

export default StarRating;
