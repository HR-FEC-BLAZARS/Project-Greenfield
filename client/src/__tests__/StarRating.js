import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount, shallow, render } from 'enzyme';
import StarRating from '../components/StarRating.jsx';

configure({ adapter: new Adapter() });

describe('StarRating', () => {
  it('should render', () => {
    const wrapper = shallow(<StarRating />);
    expect(wrapper.exists()).toBeTruthy();
  });

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<StarRating debug />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with no props', () => {
    const component = shallow(<StarRating />);

    expect(component).toMatchSnapshot();
  });
});
