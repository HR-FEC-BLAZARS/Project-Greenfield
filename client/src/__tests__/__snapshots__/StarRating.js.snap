// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`StarRating should render correctly with no props 1`] = `ShallowWrapper {}`;
